modules/python/python-logmsg.c

```diff
@@ -113,6 +113,17 @@ _py_log_message_setattr(PyObject *o, PyObject *key, PyObject *value)
   PyObject *value_as_strobj = PyObject_Str(value);
   if (value_as_strobj)
     {
+      if(log_msg_is_write_protected(py_msg->msg))
+        {
+          msg_debug("Error while modifying",
+                    evt_tag_printf("msg", "%p", py_msg),
+                    evt_tag_str("name", name),
+                    evt_tag_str("value", py_object_as_string(value_as_strobj)));
+
+          PyErr_SetString(PyExc_RuntimeError, "msg is write protected");
+          return -1;
```

Comment:
This branch causes a memory leak. We need to unref value_as_strobj

Reply:
fixed
```Py_DECREF(value_as_strobj);```